src/AsyncTestCase.php

```diff
@@ -0,0 +1,41 @@
+<?php declare(strict_types=1);
+
+/**
+ * A PHPUnit TestCase intended to help facilitate writing async tests by running each test on the amphp Loop and
+ * ensuring that the test runs until completion based on your test returning either a Promise or a Generator.
+ */
+
+namespace Amp\PHPUnit;
+
+use Amp\Coroutine;
+use Amp\Loop;
+use Amp\Promise;
+use Amp\Success;
+use PHPUnit\Framework\TestCase as PHPUnitTestCase;
+use Generator;
+
+abstract class AsyncTestCase extends PHPUnitTestCase {
+
+    public function runTest() {
+        Loop::run(function() {
```

Comment:
I'm not completely sure how PHPUnit might use the return value of a test. I plan on doing some more research and introduce a by-reference variable that will return whatever the Promise resolves with.

Reply:
So the return value of a test is used by the `@depends` annotation. https://phpunit.readthedocs.io/en/7.1/writing-tests-for-phpunit.html#writing-tests-for-phpunit-test-dependencies I have personally never used this functionality and honestly don't really see its value, seems more of a code smell to me. We could solve this by returning the value the Promise is resolved with and document it or simply take the stance that the AsyncTestCase does not support the `@depends` annotation and document it.

Reply:
IMO `@depends` should be supported and the value simply returned. We used that in several tests, but I'm not sure whether we make use of the return value. Instead of `Amp\Loop::run()`, it could use `Amp\Promise\wait()`.